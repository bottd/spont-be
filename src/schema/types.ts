import * as graphql from 'graphql';
import { selectLocationsByUserID, selectUsersByLocationID } from './utils';

const {
  GraphQLObjectType,
  GraphQLFloat,
  GraphQLList,
  GraphQLString,
  GraphQLNonNull,
} = graphql;

export const UserType = new GraphQLObjectType({
  name: 'User',
  fields: () => ({
    id: { type: GraphQLString },
    created_at: { type: GraphQLString },
    updated_at: { type: GraphQLString },
    locations: {
      type: new GraphQLList(LocationType),
      resolve(parentValue, args) {
        return selectLocationsByUserID(parentValue.id);
      },
    },
  }),
});

export const LocationType = new GraphQLObjectType({
  name: 'Location',
  fields: () => ({
    id: { type: GraphQLString },
    location_name: { type: GraphQLString },
    category: { type: GraphQLString },
    latitude: { type: GraphQLFloat },
    longitude: { type: GraphQLFloat },
    users: {
      type: new GraphQLList(UserType),
      resolve(parentValue, args) {
        return selectUsersByLocationID(args.id);
      },
    },
  }),
});

export const CoordType = new GraphQLObjectType({
  name: 'Coordinate',
  fields: () => ({
    userID: { type: GraphQLString },
    latitude: { type: GraphQLFloat },
    longitude: { type: GraphQLFloat },
  }),
});
