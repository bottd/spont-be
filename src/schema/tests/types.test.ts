import { UserType, LocationType } from '../types';

describe('types', () => {
  describe('UserType', () => {
    it('should match snapshot', () => {
      expect(UserType).toMatchSnapshot();
    });
  });
  describe('LocationType', () => {
    it('should match snapshot', () => {
      expect(LocationType).toMatchSnapshot();
    });
  });
});
