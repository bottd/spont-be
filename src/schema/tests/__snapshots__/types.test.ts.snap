// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`types LocationType should match snapshot 1`] = `"Location"`;

exports[`types UserType should match snapshot 1`] = `"User"`;
